Source/ParticleInterpolators/ParticleInterpolators.impl.hpp

```diff
@@ -0,0 +1,655 @@
+    //         m_hi_boundary_reflective[dir])
+    //         {
+    //             amrex::Abort("Help!!! reflective BCs are implemented, but not
+    //             "
+    //                          "yet tested. Likely to give absolute
+    //                          gibberish!");
+    //         }
+    //     }
+    // }
+
+    // get total query points here
+    int npts = 0;
+    if (amrex::ParallelDescriptor::IOProcessor())
+    {
+        npts = static_cast<int>(query.numPoints()); // only rank 0 touches query
+    }
+    // broadcast
+    amrex::ParallelDescriptor::Bcast(
+        &npts, 1, amrex::ParallelDescriptor::IOProcessorNumber());
+
+    // value_at_point[k][ip], where k in [0..num_components-1]
+    std::vector<std::vector<amrex::Real>> value_at_point(
+        num_components, std::vector<amrex::Real>(npts, 0.0));
+    // a vector to mark which points have values
+    std::vector<int> have(npts, 0);
+
+    int local_particle_counter = 0;
+
+    // gather all particle values
+    for (int lev = 0; lev <= this->finestLevel(); ++lev)
+    {
+        for (ParIterType it(*this, lev); it.isValid(); ++it)
+        {
+            local_particle_counter += it.numParticles();
+
+            const auto &ptile = this->ParticlesAt(lev, it);
+            const auto aos    = ptile.GetArrayOfStructs();
+            const auto *P     = aos.data();
+            const auto soa    = ptile.getConstParticleTileData();
+            const int np      = it.numParticles();
+
+            amrex::Gpu::HostVector<ParticleType> hp(np);
+            amrex::Gpu::copyAsync(amrex::Gpu::deviceToHost, aos.data(),
+                                  aos.data() + np, hp.begin());
+
+            std::vector<amrex::Gpu::HostVector<amrex::ParticleReal>> hr(
+                num_components);
+
+            for (int k = 0; k < num_components; ++k)
+            {
+                hr[k].resize(np);
+                auto dptr = ptile.GetStructOfArrays().GetRealData(k).data();
+                amrex::Gpu::copyAsync(amrex::Gpu::deviceToHost, dptr, dptr + np,
+                                      hr[k].begin());
+            }
+            amrex::Gpu::streamSynchronize();
+
+            for (int i = 0; i < np; ++i)
+            {
+                const int q = hp[i].idata(0); // get particle index
+                if (q < 0 || q >= npts)
+                {
+                    amrex::Abort("interp(): particle id out of range");
+                }
+                for (int k = 0; k < num_components; ++k)
+                {
+                    value_at_point[k][q] = static_cast<double>(hr[k][i]);
+                }
+                have[q] = 1; // mark that we have a value for this point
+            }
+        }
+    }
+    if (m_verbosity)
+    {
+        amrex::AllPrint() << "rank " << amrex::ParallelDescriptor::MyProc()
+                          << " holds " << local_particle_counter
+                          << " particles\n";
```

Comment:
Let's prefix this printing with `ParticleInterpolator` e.g.
```suggestion
        amrex::AllPrint() << "ParticleInterpolator: rank "
                          << amrex::ParallelDescriptor::MyProc()
                          << " holds " << local_particle_counter
                          << " particles\n";
```